### ISSUE TITLE
Modify build behavior

### ISSUE THREAD

**Issue by davidnuzik** [rancher/rio#627]
Allow rio up to build from a Dockerfile and have default build params.
i.e. cli can build just from local repo; don't need riofile. We construct one if none exists.

**Comment by rancher-max** [comment]
Appears to work properly. I cloned https://github.com/rancher-max/rio-demo, navigated to the directory, and ran `rio up --name noriofile`

Output:
```
default:stack/noriofile
INFO[0001] Building service rio-demo                    
[+] Building 6.5s (10/10) FINISHED                                                                                                                                                                    
 => [internal] load build definition from Dockerfile                                                                                                                                             0.3s
 => => transferring dockerfile: 269B                                                                                                                                                             0.2s
 => [internal] load .dockerignore                                                                                                                                                                0.2s
 => => transferring context: 2B                                                                                                                                                                  0.2s
 => [internal] load metadata for docker.io/library/golang:1.12.1-alpine3.9                                                                                                                       1.1s
 => [1/5] FROM docker.io/library/golang:1.12.1-alpine3.9@sha256:5f7781ceb97dd23c28f603c389d71a0ce98f9f6c78aa8cbd12b6ca836bfc6c6c                                                                 0.0s
 => => resolve docker.io/library/golang:1.12.1-alpine3.9@sha256:5f7781ceb97dd23c28f603c389d71a0ce98f9f6c78aa8cbd12b6ca836bfc6c6c                                                                 0.0s
 => [internal] load build context                                                                                                                                                                0.5s
 => => transferring context: 63.79kB                                                                                                                                                             0.5s
 => CACHED [2/5] RUN ["mkdir", "-p", "/go/src/github.com/rancher/demo"]                                                                                                                          0.0s
 => [3/5] COPY * /go/src/github.com/rancher/demo/                                                                                                                                                0.1s
 => [4/5] WORKDIR /go/src/github.com/rancher/demo                                                                                                                                                0.1s
 => [5/5] RUN ["go", "build", "-o", "demo"]                                                                                                                                                      1.5s
 => exporting to image                                                                                                                                                                           2.8s
 => => exporting layers                                                                                                                                                                          1.8s
 => => exporting manifest sha256:646fb6d45109121902b64e4dfbf05d1440fdaa757cc5cc24f9d9b9c88e7ed788                                                                                                0.0s
 => => exporting config sha256:f08daaee33316424dc4c4411b8fe8f3370c5a596721a60cea20fc8c838570fca                                                                                                  0.0s
 => => pushing layers                                                                                                                                                                            0.9s
 => => pushing manifest for localhost:80/default/rio-demo:latest  
```

**Comment by rancher-max** [comment]
Also tried this with `rio up https://github.com/rancher-max/rio-demo` which gave output: `default:stack/documents` (ran while working directory was my Documents folder). This built service successfully.

I then changed a value in the code in the repo, which did not update my service automatically. So, I ran `rio up --name diffname1 https://github.com/rancher-max/rio-demo`, which overwrite the service.
I was curious because it kept choosing the name as: `source@827eb250`, so I ran `rio up --name diffname2 https://github.com/rancher/rio-demo` (different repo). 
This also overwrote the service. So I believe there is an issue with running `rio up <gh repo>` when the repo does not contain a Riofile.
